consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java

```diff
@@ -298,17 +300,62 @@ public void run() {
-                    currentRun.setOverwrittenDocumentCount(overwriteLightblue(documentsToOverwrite));
+                    List<LightblueResponse> responses = overwriteLightblue(documentsToOverwrite);
+                    int totalOverwrittenDocumentCount = 0;
+
+                    for (LightblueResponse r : responses) {
+                        totalOverwrittenDocumentCount += r.parseModifiedCount();
+                        JsonNode jsonNode = r.getJson();
+                        JsonNode jsStatus = jsonNode.findValue("status");
+                        if (jsStatus == null || jsStatus instanceof NullNode || jsStatus instanceof MissingNode) {
+                            // It should not reach here as is expected lightblue to throw an exception
+                            LOGGER.error("In sourceDocuments: Found 'error' in response's status in one of the documents!");
+                            throw new RuntimeException("Found 'error' in response's status!");
+                        }
+                        String status = jsStatus.asText(); // TODO lightblue client should have a better way to handle this
+                        if (!STATUS_COMPLETE.equalsIgnoreCase(status)) {
+                            allComplete = false;
+                            if (STATUS_PARTIAL.equalsIgnoreCase(status)) {
+                                anyPartial = true;
+                            } else if (STATUS_ASYNC.equalsIgnoreCase(status)) {
+                                anyAsync = true;
+                            } else if (STATUS_ERROR.equalsIgnoreCase(status)) {
+                                // It should not reach here as is expected lightblue to throw an exception
+                                LOGGER.error("In sourceDocuments: Found 'error' in response's status in one of the documents!");
+                                throw new RuntimeException("Found 'error' in response's status!");
+                            } else {
+                                // It should not reach here as lightblue response should always have a status
+                                LOGGER.error("In sourceDocuments: One of the JSON returned doesn't have a valid status. Status set on the response was \"{}\"", status);
+                                throw new RuntimeException("Invalid status from the response: " + status);
+                            }
+                        }
+                    }
+                    currentRun.setOverwrittenDocumentCount(totalOverwrittenDocumentCount);
+                }
+                if (allComplete) {
+                    currentRun.setJobStatus(JobStatus.COMPLETED_SUCCESS);
+                } else if (anyPartial) {
+                    currentRun.setJobStatus(JobStatus.COMPLETED_PARTIAL);
+                } else if (anyAsync) {
+                    currentRun.setJobStatus(JobStatus.COMPLETED_IGNORED);
+                } else {
+                    currentRun.setJobStatus(JobStatus.UNKNOWN);
```

Comment:
Well I created this why so it would be very clear how the status for the migration would be influenced by the status of any of the requests (I even made int eh beginning  a more detailed way which I would count the returns and etc but it would be over engineering). So Here we can just see that if you got any partial response it would have a higher priority than if you get an Async and so on (which mean that a migration job can only have one status of time and we have to deal with different documents status  )